client/components/embed-container/index.jsx

```diff
@@ -192,6 +196,34 @@ function embedSlideshow( domNode ) {
 	}
 }
 
+function embedCarousel( domNode ) {
+	debug( 'processing carousel for ', domNode );
+
+	const carouselItemsWrapper = domNode.querySelector( '.swiper-wrapper' );
+
+	// Inject the DotPager component.
+	if ( carouselItemsWrapper ) {
+		const carouselItems = Array.from( carouselItemsWrapper?.children );
+
+		if ( carouselItems ) {
+			createRoot( domNode ).render(
+				<DotPager>
+					{ carouselItems.map( ( item, index ) => {
+						const itemProps = {
+							key: index,
+							className: classNames( 'carousel-slide', item?.className ),
+							id: index,
+							// eslint-disable-next-line react/no-danger
+							dangerouslySetInnerHTML: { __html: item?.innerHTML },
+						};
+						return <div { ...itemProps } />;
```

Comment:
This use of an object and then a spread operator for all the fields feels unexpected and unusual to me. Could you comment on why you're doing it this way? (I feel like I am missing something!)

I would normally expect something more like the following:
```suggestion
						return (
							<div
								key={ index }
								className={ classNames( 'carousel-slide', item?.className ) }
								dangerouslySetInnerHTML={ __html: item?.innerHTML }
							/>;
```

Reply:
🤦‍♀️ At one point I had refactored this to use `createElement` so we could change the main class from a DIV to whatever the original item class was, but changed my mind and forgot to switch it back to the standard format here. Fixed. :)